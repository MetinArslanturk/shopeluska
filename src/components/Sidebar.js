import React from 'react';
import { connect } from 'react-redux';
import { Layout } from 'antd';
const { Sider } = Layout;

const Sidebar = ({showSidebar}) => {
    return (
        <>
            {showSidebar && <Sider breakpoint="lg"
                collapsedWidth="0">Sidebar</Sider>
                }
        </>
    )
}

const mapStateToprops = (state) => ({
    showSidebar: state.layout.showSidebar
})
export default connect(mapStateToprops)(Sidebar)