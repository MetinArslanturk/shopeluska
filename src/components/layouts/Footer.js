import React from 'react';

const Footer = () => {
    return (
        <>
            <div>
                Â©{' '}
                <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href="https://www.metinarslanturk.com"
                >
                    Metin Arslanturk
                </a>{' '}
                <span className="mail-text">
                    ( contact@metinarslanturk.com )
                </span>{' '}
                --{' '}
                <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href="https://github.com/MetinArslanturk/shopeluska"
                >
                    GitHub of This Project
                </a>
            </div>
        </>
    );
};

export default Footer;
