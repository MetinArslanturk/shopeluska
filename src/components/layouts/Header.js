import React, { useState } from 'react';
import { connect } from 'react-redux';
import { history, baseHref } from '../../config/config';
import { Drawer, Row, Col, Button, Input, Badge } from 'antd';
import { Link } from 'react-router-dom';
import { BarsOutlined, ShoppingCartOutlined } from '@ant-design/icons';
import { startLogout } from '../../actions/auth';
import {isAuthenticated} from '../../selectors/auth';
const { Search } = Input;

const clickLogin = () => {
    history.push(baseHref + 'login');
}

const clickMyAccount = () => {
    history.push(baseHref + 'my-account');
}

export const Header = ({ isAuthenticated, startLogout}) => {
    const [visibleNav, setVisibleNav] = useState(false);

    return (
        <>
            <div className="header-wrapper">

                <Row>
                    <Col span={4}>
                        <div className="brand-logo">
                            <Link to={baseHref}><img alt="logo" className="logo-image" src={baseHref + 'logo.png'} /></Link>
                        </div>
                    </Col>
                    <Col xs={4} sm={4} md={7} lg={8} xl={12}>
                        <div className="search-box">
                            <Search placeholder="input search text" onSearch={value => console.log(value)} enterButton />
                        </div>
                    </Col>
                    <Col xs={4} sm={6} md={6} lg={6} xl={4}>
                        <div className="user-nav">
                        <Badge className="badge-text" count={10}>
                        <ShoppingCartOutlined style={{ fontSize: '36px'}}/>
                        </Badge>
                        
                            {!isAuthenticated && (
                                <Button type="primary" className="login-button" onClick={clickLogin}>Login</Button>
                            )}
                            {isAuthenticated && (
                                <>
                                    <Button className="my-account-button my-account" type="primary" onClick={clickMyAccount}>My Account</Button>
                                    <Button className="logout-button" onClick={startLogout}>Logout</Button>
                                </>
                            )}
                        </div>
                        <div className="head-menu-toggle-button">
                            <BarsOutlined onClick={e => setVisibleNav(true)} style={{ fontSize: '32px', marginTop: '18px' }} />
                        </div>
                    </Col>


                </Row>
            </div>
            <Drawer
                title="Shopeluska Navigation"
                placement="right"
                closable={true}
                onClose={() => {
                    setVisibleNav(false);
                }}
                visible={visibleNav}
            >
                <p>Some contents...</p>
                <p>Some contents...</p>
                <p>Some contents...</p>
            </Drawer>
        </>
    )
}
const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user)
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);