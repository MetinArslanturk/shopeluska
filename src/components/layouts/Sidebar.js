import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Layout } from 'antd';
import { NavLink } from 'react-router-dom';
import { baseHref } from '../../config/config';
import { menuItems  } from "../../helpers/menuItems";

const { Sider } = Layout;

const Sidebar = ({ showSidebar }) => {

    return (
        <>
            {showSidebar && (
                <Sider breakpoint="lg" width={230} collapsedWidth="0">
                    <nav className="nav">
                        <ul>
                            {menuItems.map((item) => {
                                return (
                                    <li key={item.content}>
                                        <NavLink
                                            activeClassName="active"
                                            exact
                                            to={baseHref + item.link}
                                        >
                                            {item.content}
                                        </NavLink>
                                    </li>
                                );
                            })}
                        </ul>
                    </nav>
                </Sider>
            )}
        </>
    );
};

Sidebar.propTypes = {
    showSidebar: PropTypes.bool,
};

const mapStateToprops = (state) => ({
    showSidebar: state.layout.showSidebar,
});
export default connect(mapStateToprops)(Sidebar);
