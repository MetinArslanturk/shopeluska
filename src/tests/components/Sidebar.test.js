import React from 'react';
import { shallow } from 'enzyme';
import { Sidebar } from '../../components/layouts/Sidebar';

test('should render Sidebar', () => {
    const wrapper = shallow(<Sidebar showSidebar />);
    expect(wrapper).toMatchSnapshot();
});
