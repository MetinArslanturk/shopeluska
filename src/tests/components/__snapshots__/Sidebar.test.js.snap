// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar 1`] = `
<Fragment>
  <Sider
    breakpoint="lg"
    collapsedWidth="0"
    width={230}
  >
    <nav
      className="nav"
    >
      <ul>
        <li
          key="Homepage"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/shopeluska/"
          >
            Homepage
          </NavLink>
        </li>
        <li
          key="Clothing"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/shopeluska/category/clothing"
          >
            Clothing
          </NavLink>
        </li>
        <li
          key="Shoes"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/shopeluska/category/shoes"
          >
            Shoes
          </NavLink>
        </li>
        <li
          key="Accessories"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/shopeluska/category/accessories"
          >
            Accessories
          </NavLink>
        </li>
      </ul>
    </nav>
  </Sider>
</Fragment>
`;
